slides/week3/git.org

```diff
@@ -0,0 +1,200 @@
+#+TITLE: Git Training - Week 3
+#+AUTHOR: Jay Kamat, Josh Ting
+#+EMAIL: jaygkamat@gmail.com, josh.ting@gatech.edu
+
+* What You Know
+- ~git status~
+- ~git add~
+- ~git commit~
+- ~git remote~
+- ~git pull~
+- ~git push~
+- /(any questions?)/
+
+* Some Problems
+- Multiple developers work on the same code, at the same time
+- We want to maintain multiple versions of the same software in parallel
+  - How do I apply a security fix to an old version?
+- You want a safe place to test out changes, without affecting anyone else
+- Branching is ~git~'s solution
+
+* Branches
+- Branches are local copies of state
+- If you do work on a branch, it won't affect any other branches
+- Later on, we can 'merge' branches together (so work done on either can be combined).
+- Branches are what people have the most trouble with
+  - _*Ask questions /please/!*_ This is a difficult topic.
+
+** Git As a Graph
+
+- git history is a *graph*
+- Each commit is a node on the graph
+- Ways to see it as a graph, rather than a line of commits
+  - ~git~ porcelain
+    #+BEGIN_SRC sh
+      git log --oneline --graph --all
+    #+END_SRC
+
+    #+BEGIN_SRC sh :results output :exports results
+      git log --oneline --graph | head -n10
+    #+END_SRC
+  - ~gitk~
+  - ~git~ integrations for your favorite editor!
+
+** Creating Branches
+- Let's start by creating multiple branches
+
+  #+BEGIN_SRC
+    844a162 * master stable beta Initial commit
```

Comment:
I thought it looked something like that, but `git` on my machine doesn't format output like that...
```
kristaps@KB-XPS15:/mnt/c/Users/Kristaps/Documents/GitHub/jay-software-training$ git log --oneline --graph --all
* 28ad8c5 Slides to make pr
* 183dba8 Branching edits
*   2771727 Merge branch 'weekThreeGitSlides' of https://github.com/jgkamat/software-training into weekThreeGitSlides
|\
| * 16f86c1 Add export of git log
| * 88407a2 Add first draft of week 3 git slides
* |   f3af921 Merge pull request #27 from jgkamat/jay/vendor-reveal
|\ \
| * | f4ef4e9 Attempt to fix old artifacts staying in gh-pages
| * | 5a357df Update reveal link to point to RJ fork
|/ /
* |   3652c31 Merge pull request #21 from RoboJackets/week1fix
|\ \
| |/
|/|
| *   32f0cc9 Merge branch 'master' into week1fix
| |\
| |/
```